### ISSUE TITLE
When using Diode, Fatal doesn't output anything

### ISSUE THREAD

**Issue by reflog** [rs/zerolog#324]
Here's a small test to demonstrate:
```
package main

import (
	"fmt"
	"os"

	"github.com/rs/zerolog"
	"github.com/rs/zerolog/diode"
)

func main() {
	w := diode.NewWriter(os.Stdout, 1000, 0, func(missed int) {
		fmt.Printf("Dropped %d messages\n", missed)
	})
	log := zerolog.New(w)
	log.Print("this will print")
	log.Fatal().Msg("this will not")
	w.Close()
}
```

https://play.golang.org/p/pYkYlBrde_7

**Comment by cesher** [comment]
Fatal calls `panic()` internally, that won't give diode enough time to flush the buffer.

To work around this in my project I use `logger.WithLevel(zerolog.FatalLevel)`.

**Comment by gmbuell** [comment]
FYI this is fixed in #634
